addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/ProxyDisclosureScanRuleUnitTest.java

```diff
@@ -57,4 +68,33 @@ void shouldReturnExpectedMappings() {
                 tags.get(CommonAlertTag.OWASP_2017_A06_SEC_MISCONFIG.getTag()),
                 is(equalTo(CommonAlertTag.OWASP_2017_A06_SEC_MISCONFIG.getValue())));
     }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "X-Forwarded-For: 76.69.54.171", "X-Forwarded-For: 127.0.0.1",
+                "X-Forwarded-Host: api.test.glaypen.garnercorp.com", "X-Forwarded-Port: 443",
```

Comment:
Better remove public IPs and domains, there are also headers being tested that wouldn't raise an alert anyway. It's missing Via (for completeness).

Reply:
The test is still not testing the expected code.

Reply:
Sorry I glossed over your second point about the served content. Will adjust.

Reply:
I do plan to work through this, but it's gonna take a bit longer to get the nano handler setup for all the pre-checks.